### ISSUE TITLE
[BUG] Residuals with val_loss 

### ISSUE THREAD

**Issue by martinb-bb** [unit8co/darts#1155]
**Describe the bug**
Trying to calculate residuals with `val_loss`.
Is there a reason we cannot use it and are forced to use `train_loss`?

![image](https://user-images.githubusercontent.com/105685594/184882365-48783d65-2809-4df9-a07b-25cb81bedc25.png)

Thank you!

**Comment by dennisbader** [comment]
`val_loss` is only available when passing a validation set to `model.fit()` (i.e., with `val_series`, `val_past_covariates`, ...)

**Comment by martinb-bb** [comment]
We are already doing that:
<img width="463" alt="image" src="https://user-images.githubusercontent.com/105685594/184906077-b2b72e22-3721-4c25-9f8f-047795f89cdf.png">


We are already doing that hence why we can track our `val_loss`. Could you let me know why residuals does not allow for us to use `val_loss`.

Am I missing something?

**Comment by martinb-bb** [comment]
Hi @dennisbader  Thanks for the information. 
We are already training our models with `val_series` and `val_past_covariates`. 


Then to calculate residuals: 

<img width="584" alt="image" src="https://user-images.githubusercontent.com/105685594/184909827-00130c4b-14ea-4ade-82c5-8509cf46eb00.png">

But then complains that we are checking `val_loss` when running `model.residuals()` 
<img width="1003" alt="image" src="https://user-images.githubusercontent.com/105685594/184910028-9ece105a-91f7-490c-85d3-ff3dd6303af3.png">

It is checking how the original model was trained. If using `val_loss` then it will break residuals. Residuals will only work if i train my original model with `train_loss`. Is my understanding correct?

**Comment by dennisbader** [comment]
Ah, now I understand. `model.residuals()` doesn't accept a validation set. This is because only our GlobalForecastingModels support the validation set in the `fit()` function (and LocalForecastingModels do not but share the same API). 

To compute the residuals you have to train the model first and then  set `model.residuals(..., retrain=False)`.

Some things to add:
- `model.pl_trainer_kwargs = pl_trainer_kwargs` does not have any effect. Instead do `model.trainer_params = pl_trainer_kwargs`.
- It is recommended to add the trainer kwargs at model creation with `pl_trainer_kwargs`

**Comment by martinb-bb** [comment]
Ah that makes sense! Thanks. Totally understand. One other weird thing:

Are you able to set the retrain flag on your end?
<img width="368" alt="image" src="https://user-images.githubusercontent.com/105685594/184916926-6b0be6af-e583-4b2f-b492-bc6ca6859b4e.png">

<img width="477" alt="image" src="https://user-images.githubusercontent.com/105685594/184917104-f6b937d0-a0d6-49ee-815d-d8a8ba2d7b78.png">

**Comment by dennisbader** [comment]
No worries! We introduced `retrain` to `residuals()` in the latest release darts==0.21.0.
If you upgrade your darts version it should work.
